components/layout/query.rs

```diff
@@ -37,16 +40,111 @@ use style::values::specified::GenericGridTemplateComponent;
 use style::values::specified::box_::DisplayInside;
 use style::values::specified::text::TextTransformCase;
 use style_traits::{ParsingMode, ToCss};
+use webrender_api::units::LayoutTransform;
 
 use crate::ArcRefCell;
 use crate::dom::NodeExt;
 use crate::flow::inline::construct::{TextTransformation, WhitespaceCollapse, capitalize_string};
 use crate::fragment_tree::{
     BoxFragment, Fragment, FragmentFlags, FragmentTree, SpecificLayoutInfo,
 };
+use crate::style_ext::TransformExt;
 use crate::taffy::SpecificTaffyGridInfo;
 
-pub fn process_content_box_request(node: ServoLayoutNode<'_>) -> Option<Rect<Au>> {
+/// Specific helper for post composite query, query that requires the consideration of transform,
+/// mapping of coordinates, scroll offset, etc. It provide some encapsulation of borrowed structures
+/// required in these kind of queries and some useful utilities of it.
+pub(crate) struct PostCompositeQueryHelper<'dom> {
+    scroll_tree: Option<&'dom ScrollTree>,
+}
+
+impl<'dom> PostCompositeQueryHelper<'dom> {
+    pub fn new(scroll_tree: Option<&'dom ScrollTree>) -> Self {
+        PostCompositeQueryHelper { scroll_tree }
+    }
+
+    fn retrieve_box_fragment(fragment: &Fragment) -> Option<&ArcRefCell<BoxFragment>> {
+        match fragment {
+            Fragment::Box(box_fragment) | Fragment::Float(box_fragment) => Some(box_fragment),
+            _ => None,
+        }
+    }
```

Comment:
I would add this to `Fragment` and call it `fn box_fragment`. It's a bit weird to have the helper here.